src/ApiClient/util/AuthController.ts

```diff
@@ -139,37 +157,43 @@ export default new (class AuthController {
                 });
             }
         } catch (err) {
-            return new InternalStatus<null, LoginErrors>({
+            const stat = new InternalStatus<null, LoginErrors>({
                 code: StatusCode.ERROR,
                 error: err as InternalError<LoginErrors>
             });
-        } finally {
-            this.loggingIn = false;
-            this.loggingInDoneEvent.emit("authend");
+            this.emit("loggingInDoneEvent", stat);
+            return stat;
         }
 
         if (!response || !response.data) {
-            return new InternalStatus<null, LoginErrors>({
+            const stat = new InternalStatus<null, LoginErrors>({
                 code: StatusCode.ERROR,
                 error: new InternalError(ErrorCode.APP_FAIL, {
                     jsError: new Error("unknown credential type")
                 })
             });
+            this.emit("loggingInDoneEvent", stat);
+            this.loggingIn = false;
+            return stat;
```

Comment:
That's alot of duplicated code, move the variable definition for stat outside and return it once at the end